public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx

```diff
@@ -0,0 +1,134 @@
+import * as H from 'history';
+
+import { locationService } from '@grafana/runtime';
+import {
+  getUrlSyncManager,
+  SceneFlexItem,
+  SceneFlexLayout,
+  SceneObject,
+  SceneObjectBase,
+  SceneObjectRef,
+  SceneObjectState,
+  sceneUtils,
+  SplitLayout,
+  VizPanel,
+} from '@grafana/scenes';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { getDashboardUrl } from '../utils/utils';
+
+import { PanelEditorRenderer } from './PanelEditorRenderer';
+import { PanelOptionsPane } from './PanelOptionsPane';
+
+export interface PanelEditorState extends SceneObjectState {
+  body: SceneObject;
+  controls?: SceneObject[];
+  isDirty?: boolean;
+  /** Panel to inspect */
+  inspectPanelId?: string;
+  /** Scene object that handles the current drawer */
+  drawer?: SceneObject;
+
+  dashboardRef: SceneObjectRef<DashboardScene>;
+  sourcePanelRef: SceneObjectRef<VizPanel>;
+  panelRef: SceneObjectRef<VizPanel>;
+}
+
+export class PanelEditor extends SceneObjectBase<PanelEditorState> {
+  static Component = PanelEditorRenderer;
+
+  public constructor(state: PanelEditorState) {
+    super(state);
+
+    this.addActivationHandler(() => this._activationHandler());
+  }
+
+  private _activationHandler() {
+    // Deactivation logic
+    return () => {
+      getUrlSyncManager().cleanUp(this);
+    };
+  }
+
+  public startUrlSync() {
+    getUrlSyncManager().initSync(this);
+  }
+
+  public getPageNav(location: H.Location) {
+    return {
+      text: 'Edit panel',
+      parentItem: this.state.dashboardRef.resolve().getPageNav(location),
+    };
+  }
+
+  public onDiscard = () => {
+    // Open question on what to preserve when going back
+    // Preserve time range, and variables state (that might have been changed while in panel edit)
+    // Preserve current panel data? (say if you just changed the time range and have new data)
```

Comment:
Think we should replicate the current behavior first, I mean preserving whatever "global" controls have changed. This means data for a panel ideally not refreshed, but all other panels should exec new queries.

Reply:
@dprokop yea, I agree